routers/routes/routes.go

```diff
@@ -857,6 +857,8 @@ func RegisterRoutes(m *macaron.Macaron) {
 			m.Get("/?:page", repo.Wiki)
 			m.Get("/_pages", repo.WikiPages)
 			m.Get("/:page/_revision", repo.WikiRevision)
+			m.Get("/commit/:sha([a-f0-9]{7,40})$", repo.SetEditorconfigIfExists, repo.SetDiffViewStyle, repo.Diff)
```

Comment:
I would say that if the intention is that only commits are viewable here that this does not ensure this.

A good example is that I can create the branch `deadbeef` and this will be rendered as commit. (Passing a tree sha doesn't seem to work though - which is good.)

Reply:
Hmm I just copied from the existing router for repos -- I suppose my only intention was that the link from the wiki history works.

I don't think it matters that you could manually type that in and see a commit since you would have permission to view it anyway, but would there be a simple way to prevent that?

Reply:
I'm not certain there is a clean way tbh - even appending ^{commit} doesn't prevent git rev-parse parsing a reference. (I recognized that they were a copy but I just thought I'd better alert people to this point.)

The only solution is to rev-parse it and ensure that what comes back starts with what was put in?